tests/lightning/test_dist_ckpt.py

```diff
@@ -0,0 +1,161 @@
+import os
+import types
+from pathlib import Path
+from typing import Optional, Tuple
+
+import pytest
+import pytorch_lightning as pl
+import torch
+
+import nemo.lightning as nl
+from nemo.collections import llm
+from nemo.lightning.io.pl import MegatronCheckpointIO
+from nemo.utils.callbacks.dist_ckpt_io import AsyncFinalizableCheckpointIO, AsyncFinalizerCallback
+
+
+def _get_nlp_strategy_without_optimizer_state():
+    strategy = nl.MegatronStrategy(
+        enable_nemo_ckpt_io=False,
+    )
+    # this ensures optimizer sharded state creation is skipped
```

Comment:
Did you test if with the new strategy testing the optimizer save works?

Previously it errored because of some parallel state misconfiguration I believe

Reply:
The tests still pass if I allow the optimizer state to be created. Is that sufficient?

Reply:
yes thanks